packages/shared/src/hooks/useFeedLayout.ts

```diff
@@ -1,45 +1,67 @@
-import { useMemo } from 'react';
-import { SharedFeedPage } from '../components/utilities';
-import { AllFeedPages } from '../lib/query';
-import { useActiveFeedNameContext } from '../contexts/ActiveFeedNameContext';
 import { useViewSize, ViewSize } from './useViewSize';
+import { useActiveFeedNameContext } from '../contexts/ActiveFeedNameContext';
+import { FeedPage, FeedPageLayoutMobile } from '../components/utilities';
+import { AllFeedPages } from '../lib/query';
+
+interface UseFeedLayoutReturn {
+  shouldUseMobileFeedLayout: boolean;
+  FeedPageLayoutComponent: React.ComponentType;
+}
 
 interface UseFeedLayoutProps {
-  feedName?: AllFeedPages;
   feedRelated?: boolean;
 }
 
-interface UseFeedLayout {
-  shouldUseMobileFeedLayout: boolean;
+export enum FeedPagesWithMobileLayout {
+  MyFeed = 'my-feed',
+  Popular = 'popular',
+  Search = 'search',
+  Upvoted = 'upvoted',
+  Discussed = 'discussed',
+  Bookmarks = 'bookmarks',
+  UserPosts = 'user-posts',
+  UserUpvoted = 'user-upvoted',
+  SquadPage = 'squads[handle]',
+  TagPage = 'tags[tag]',
+  SourcePage = 'sources[source]',
+}
+
+export enum UserProfileFeed {
+  UserPosts = 'user-posts',
+  UserUpvoted = 'user-upvoted',
 }
 
 export const FeedLayoutMobileFeedPages = new Set(
-  Object.values(SharedFeedPage).filter(
-    (feedPage) => feedPage !== SharedFeedPage.Search,
-  ),
+  Object.values(FeedPagesWithMobileLayout),
 );
 
-const checkShouldUseMobileFeedLayout = (feedName: SharedFeedPage): boolean =>
-  FeedLayoutMobileFeedPages.has(feedName) && feedName !== SharedFeedPage.Search;
+export const UserProfileFeedPages = new Set(Object.values(UserProfileFeed));
+
+const checkShouldUseMobileFeedLayout = (
+  isLaptop: boolean,
+  feedName: AllFeedPages | UserProfileFeed | FeedPagesWithMobileLayout,
+): boolean =>
+  (!isLaptop &&
+    FeedLayoutMobileFeedPages.has(feedName as FeedPagesWithMobileLayout)) ||
+  UserProfileFeedPages.has(feedName as UserProfileFeed);
```

Comment:
Change from last PR that broke some feeds where this was totally removed, have reintroduced this, so now it works as it was working before where everything was fine, and have added the new feeds we need in the enums

Reply:
So this will exclude public squads and notifications basically as they are not described?

Not for now, but maybe we should have a all feeds enum and abstract portions of it to make it more clear what types of feed exist.

Reply:
> So this will exclude public squads and notifications basically as they are not described?

Exactly, before I removed this check totally since I didn't consider the squads public page and notifications as feed since they had different components, but they were using the same layout so that's where the issue happened.

> Not for now, but maybe we should have a all feeds enum and abstract portions of it to make it more clear what types of feed exist.

We have this at the moment [OtherFeedPages](https://github.com/dailydotdev/apps/blob/9e70f98d53203b0da37e1025cb3f853aad35b4c9/packages/shared/src/lib/query.ts#L16-L31) + [Shared feed pages ](https://github.com/dailydotdev/apps/blob/62c027157da0e8fc778044bbcda3149c4babfe31/packages/shared/src/components/utilities/common.tsx#L162-L168) which we usually use everywhere. 

And in this PR I created this for [mobileLayout](https://github.com/dailydotdev/apps/blob/7aeff0aa7442b6699a2c20733e85f2842ac61951/packages/shared/src/hooks/useFeedLayout.ts#L15-L27) pages. I couldn't extract portion of the enums unless I convert them into an array or sth, so went with this way. Let me know if it's best to extract what I need from the enums to an array to use.